### PR TITLE
Refacto/corporation

### PR DESCRIPTION
**Pull request by Mal3kit**
Change corporation service : now using entity.
Change resource for using mapper.
Create some unit tests and change integration test.

### CODE REVIEW

File: src/main/java/com/corpo/coliseum/api/resource/CorporationResource.java [hunk 1]
```diff
@@ -24,14 +34,35 @@
...
+    }
+
+    @GET
+    @Path("/register")
+    public Response register(@QueryParam("corporationName") String corporationName,
+                             @QueryParam("userMail") String userMail) throws ModelNotFoundException, UserException {
+        corporationService.register(corporationName, userMail);
+        return Response.ok().build();
     }
 }
```
Comment (Mal3kit) [inline src/main/java/com/corpo/coliseum/api/resource/CorporationResource.java:68]: Add method to find corporation by name

File: src/main/java/com/corpo/coliseum/api/resource/CorporationResource.java [hunk 2]
```diff
@@ -24,14 +34,35 @@
...
 
     @POST
-    public Response create(@QueryParam String name, @QueryParam String sport) {
-        corporationService.create(name, sport);
-        return Response.status(201).build();
+    public Response create(@Valid CorporationDTO corporationDTO, @Context UriInfo uriInfo) {
+        corporationService.create(modelMapper.map(corporationDTO, Corporation.class));
+        UriBuilder uriBuilder = uriInfo.getAbsolutePathBuilder().path(corporationDTO.getName());
+        return Response.created(uriBuilder.build()).build();
+    }
```
Comment (Mal3kit) [inline src/main/java/com/corpo/coliseum/api/resource/CorporationResource.java:53]: Utility of uri builder ?

File: src/main/java/com/corpo/coliseum/domain/exception/ModelNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.corpo.coliseum.domain.exception;
+
+public class ModelNotFoundException extends Exception{
+
+    public ModelNotFoundException(String message) {
+        super(message);
+    }
+
+}
```
Comment (Mal3kit) [inline src/main/java/com/corpo/coliseum/domain/exception/ModelNotFoundException.java:9]: Add Class to constructor

File: src/test/java/com/corpo/coliseum/domain/service/CorporationServiceTestIT.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Test
+    @Order(3)
+    public void testDelete() throws ModelNotFoundException {
+        corporationService.remove("Corporation 3");
+        assertEquals(2, corporationService.getAll().size());
+    }
+
+}
```
Comment (Mal3kit) [inline src/test/java/com/corpo/coliseum/domain/service/CorporationServiceTestIT.java:62]: Do only one method who call all tests

File: src/test/java/com/corpo/coliseum/domain/service/UserServiceTest.java [hunk 5]
```diff
@@ -41,5 +41,5 @@ public void testAlreadySignUp() throws UserException {
         Assertions.assertThrows(UserException.class, () -> {
             userService.signUp(USER_MAIL, "test2");
         });
-    }
+    }*/
 }
```
Comment (Mal3kit) [inline src/test/java/com/corpo/coliseum/domain/service/UserServiceTest.java:45]: Rework user tests

**Review by jeankev — APPROVED** [review]

**Pull request merged**